src/Types/GroupMetadata.ts

```diff
@@ -4,6 +4,10 @@ export type GroupParticipant = (Contact & { isAdmin?: boolean, isSuperAdmin?: bo
 
 export type ParticipantAction = 'add' | 'remove' | 'promote' | 'demote'
 
+export type RequestJoinAction = 'created' | 'revoked'
```

Comment:
there's three different actions, not only two (i'll also suggest another wording for these actions):
- `requested`: when the user sends a join request
- `retracted`: user removed their own join request
- `denied`: a group admin denied a join request

Reply:
i used created and revoked because WhatsApp calls them that, I think it's better to leave it that way, and I'll add deined

Reply:
whatsapp only have a stubtype for created 😢

Reply:
applied